### PR TITLE
Clean up cluster-local-gateway

### PR DESCRIPTION
**Pull request by nak3**
https://github.com/knative-sandbox/net-istio/pull/411 replaced cluster-local-gateway with knative-local-gateway.
Hence this patch updates the docs.

Part of https://github.com/knative/docs/issues/3271.

/cc @JRBANCEL @ZhiminXiang @arturenault

### CODE REVIEW

**Review by omerbensaadon — APPROVED** [review]
From a clarity perspective (non-technical) this /lgtm

**Review by arturenault — APPROVED** [review]
/lgtm

**Review by JRBANCEL — APPROVED** [review]

File: docs/install/installing-istio.md [hunk 1]
```diff
@@ -128,11 +128,11 @@ spec:
 EOF
 ```
 
-After you install the cluster local gateway, your service and deployment for the local gateway are both named `cluster-local-gateway`.
+After you install the cluster local gateway, your service and deployment for the local gateway is named `knative-local-gateway`.
```
Comment (csantanapr) [inline docs/install/installing-istio.md:131]: Thank you for the fix I saw a person on Slack having issues related to this

**Review by csantanapr — APPROVED** [review]
Thanks

**Pull request merged**